source/snes9x/memmap.cpp

```diff
@@ -3244,6 +3244,13 @@ void CMemory::Map_SPC7110HiROMMap (void)
 	map_hirom_offset(0xc0, 0xcf, 0x0000, 0xffff, CalculatedSize, 0);
 	map_index(0xd0, 0xff, 0x0000, 0xffff, MAP_SPC7110_ROM,  MAP_TYPE_ROM);
 
+	// For Tengai Makyou (English) 
+	//if (ROMCRC32 == 0xE589FB4)
+	if (strncmp((char*)&Memory.ROM [0xffc0], "HU TENGAI MAKYO ZERO ", 21) == 0 && CalculatedSize > 5242880)
+	{
+		map_hirom_offset(0x40, 0x4f, 0x0000, 0xffff, CalculatedSize, 0x600000);
+	}
+
```

Comment:
Cool patch. What does it do? Where'd you get these offsets from?

Reply:
@dborth it starts the rom with a different offset. Tenkai Makyou Zero starts the game with a SPC7110 version 2.2 program check if all. Necessary hardware is there. Because this is emulation it needs to know the offset.

I backported it from the following website:
https://github.com/bubble2k16/snes9x_3ds/commit/cfafe37282458779fb16c7ebd58e2dffea1f9971

Reply:
In the higher snes9x cores the offset seems to be calculated thats why it works without a patch.